mustang_codegen/lib/src/batch_generator/enum_param_visitor.dart

```diff
@@ -0,0 +1,73 @@
+import 'package:analyzer/dart/constant/value.dart';
+import 'package:analyzer/dart/element/element.dart';
+import 'package:analyzer/dart/element/visitor.dart';
+import 'package:mustang_codegen/src/batch_generator/batch_model.dart';
+import 'package:mustang_core/mustang_core.dart';
+import 'package:source_gen/source_gen.dart';
+
+class EnumParamVisitor extends SimpleElementVisitor<void> {
+  EnumParamVisitor(
+    this.serviceClassElement,
+    this.batchEnum,
+    this.fields,
+  );
+
+  final ClassElement serviceClassElement;
+
+  final EnumElement batchEnum;
+
+  List<BatchModel> fields;
+
+  @override
+  void visitMethodElement(MethodElement element) {
+    if (element.parameters.isNotEmpty) {
+      throw InvalidGenerationSourceError(
```

Comment:
Given this restriction, we cannot just add @BatchRun to any method. Probably we may have to define another method without args and annotate. Is this accurate?

Reply:
For methods annotated with `@batchRun` the guideline should be to only access data from the store and never expect the caller to provide it.